### PR TITLE
[ruff] maybe eliminate PLW2901

### PR DESCRIPTION
**Pull request by danking**
what do you think of it?

---

We should at least keep the fix to list_batches.py. I am unsure why I discovered blatant type errors in ci.

### CODE REVIEW

File: batch/batch/worker/worker.py [hunk 1]
```diff
@@ -1177,9 +1177,11 @@ def _mounts(self, uid, gid):
...
         volumes = self.image.image_config['Config']['Volumes']
         if volumes:
-            for v_container_path in volumes:
-                if not v_container_path.startswith('/'):
-                    v_container_path = '/' + v_container_path
+            for _v_container_path in volumes:
+                if not _v_container_path.startswith('/'):
+                    v_container_path = '/' + _v_container_path
+                else:
+                    v_container_path = _v_container_path
```
Comment (daniel-goldstein) [inline batch/batch/worker/worker.py:1184]: This feels like a bug, no? the `destination` that we set in the container should be as specified by the image manifest, not altered to prepend a `/`. This prepending a slash is only for putting it somewhere easy to find on the host.

I think I only added `VOLUME` support to get buildkit working which I presume uses an absolute path which is why we never noticed it, but I presume there's nothing wrong with specifying a `VOLUME` that is relative to the `WORKDIR`.

**Review by daniel-goldstein — CHANGES_REQUESTED** [review]
I just looked at your second commit, but you should be able to merge main now. Overall on board with attacking these ignored rules.

File: build.yaml [hunk 2]
```diff
@@ -2431,6 +2431,8 @@ steps:
       inline: |
         FROM {{ hail_ubuntu_image.image }}
         VOLUME ["/data"]
+        WORKDIR "/tmp"
+        VOLUME ["relative_volume]
```
Comment (daniel-goldstein) [inline build.yaml:2435]:
```suggestion
        VOLUME ["relative_volume"]
```

**Review by daniel-goldstein — CHANGES_REQUESTED** [review]
Just checking, does this use of a relative path replicate in docker? My stipulation earlier was what *seemed* reasonable, I do not remember off the top of my head the `VOLUME` spec.

**Comment by danking** [comment]
Fixed and test updated.

**Review by daniel-goldstein — APPROVED** [review]

**Pull request merged**